### PR TITLE
feat(datasets): Delete examples

### LINKED ISSUE #3349: [datasets][ui] delete dataset examples UI

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #3349
Screenshot 2024-05-30 at 9 43 08 PM

### CODE REVIEW

File: app/src/pages/datasets/DatasetActionMenu.tsx [hunk 1]
```diff
@@ -56,13 +56,6 @@ export function DatasetActionMenu(props: DatasetActionMenuProps) {
         onError: (error) => {
           onDatasetDeleteError(error);
         },
-        updater: (store) => {
```
Comment (mikeldking) [inline app/src/pages/datasets/DatasetActionMenu.tsx:59]: figured it out. Need to pass a fetch policy to refetch

Comment (axiomofjoy) [reply app/src/pages/datasets/DatasetActionMenu.tsx:59]: Nice. Will this approach fix the manual refetches we do on project deletion, too?

File: app/src/pages/project/SpansTable.tsx [hunk 2]
```diff
@@ -230,6 +230,15 @@ export function SpansTable(props: SpansTableProps) {
...
       maxSize: 10,
+      header: ({ table }) => (
+        <IndeterminateCheckboxCell
+          {...{
+            checked: table.getIsAllRowsSelected(),
+            indeterminate: table.getIsSomeRowsSelected(),
+            onChange: table.getToggleAllRowsSelectedHandler(),
+          }}
+        />
+      ),
```
Comment (axiomofjoy) [inline app/src/pages/project/SpansTable.tsx:241]: Small thing. It might be worth hiding the header checkbox when the table is empty.

**Review by axiomofjoy — APPROVED** [review]

File: app/src/pages/datasets/DatasetActionMenu.tsx (hunk 1 above)
Comment (mikeldking) [reply app/src/pages/datasets/DatasetActionMenu.tsx:59]: I don't think so

File: app/src/pages/project/SpansTable.tsx (hunk 2 above)
Comment (mikeldking) [reply app/src/pages/project/SpansTable.tsx:241]: Eh, punt ;) It's not checkable anyways

**Pull request merged**